### ISSUE TITLE
xcode generator fails when using files object

### ISSUE THREAD

**Issue by theothermatt** [mesonbuild/meson#589]
When using a files object (created with files('src.c',...)), the xcode generator fails with the following error:

```
Traceback (most recent call last):
  File "/usr/local/Cellar/meson/0.31.0/lib/python3.5/site-packages/mesonbuild/mesonmain.py", line 254, in run
    app.generate()
  File "/usr/local/Cellar/meson/0.31.0/lib/python3.5/site-packages/mesonbuild/mesonmain.py", line 158, in generate
    g.generate(intr)
  File "/usr/local/Cellar/meson/0.31.0/lib/python3.5/site-packages/mesonbuild/backend/xcodebackend.py", line 88, in generate
    self.generate_pbx_build_file()
  File "/usr/local/Cellar/meson/0.31.0/lib/python3.5/site-packages/mesonbuild/backend/xcodebackend.py", line 234, in generate_pbx_build_file
    idval = self.buildmap[s]
KeyError: 'tests/fileA.c'
```

I've set up a sample repo, which this error is from, at https://github.com/theothermatt/mesontest.
In this example, `fileA.c` is not in the `tests` directory as the error suggests, it's in the `src` directory, where a `meson.build` file creates a files object.

This works when using the ninja or vs2010 backends.

Meson version:  0.31.0

**Comment by jpakkane** [comment]
Unfortunately the XCode backend lags a bit behind the other two. Mostly because its file format is awful enough to make your eyes bleed blood.

**Comment by nirbheek** [comment]
… but also because no one has stepped up to maintain it. @nioncode has been working on the Visual Studio backends and they're in much better shape in comparison. So basically, help is appreciated for better health of this backend!

**Comment by mensinda** [comment]
Fixed with #7813.
